src/transports/smart.c

```diff
@@ -279,30 +279,31 @@ static int git_smart__connect(
 		return error;
 
 	/* Detect capabilities */
-	if (git_smart__detect_caps(first, &t->caps, &symrefs) < 0) {
-		free_symrefs(&symrefs);
-		return -1;
+	if ((error = git_smart__detect_caps(first, &t->caps, &symrefs)) == 0) {
```

Comment:
This change is wrong. We're new erroring out when the function ran successfully. It's getting fixed by the next commit, though